ivy/data_classes/array/array.py

```diff
@@ -160,7 +161,7 @@ def _init(self, data, dynamic_backend=None):
         self._dev_str = None
         self._pre_repr = None
         self._post_repr = None
-        self._backend = ivy.backend
+        self._backend = _get_backend_for_arg(self.data.__class__.__module__).backend
```

Comment:
I can't really think of a case when an `ivy.Array` is created with a different backend than the one set currently, not sure if that's possible either tbh (as ideally it would throw an error), could you please share an example of the issue being fixed? Thanks!

Reply:
it's possible to create an array with another backend than the set one using `with_backend`, but you are right, it still returns the correct backend then, reverting this

Reply:
sounds good, thanks!